codec/codec.go

```diff
@@ -21,32 +21,32 @@ type (
 
 	BinaryCodec interface {
 		// Marshal returns binary encoding of v.
-		Marshal(o ProtoMarshaler) ([]byte, error)
+		Marshal(o interface{}) ([]byte, error)
```

Comment:
preliminary review - with @aaronc, we discussed to still keep the old proto.Message interface (the one in gogoproto) because our protov2 types implement that, and then try to promoto a legacy proto.Message to the new proto.Message interface (the one with protoreflect)

Reply:
nvm didn't see aaron  already pointed that out! :D